### PR TITLE
feature/preparation-kit

### PR DESCRIPTION
**Pull request by PCOffline**
Add Preparation Kit Page

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,29 +4,30 @@
   "private": true,
```
Comment (carmeltsadok) [inline package.json:4]: Writing it here, but make sure you fix all conflicts before opening a pr :)

File: src/pages/PreparationKit/index.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import React, { useState, useLayoutEffect as useEffect, useRef } from 'react';
+import { Document, Page } from 'react-pdf/dist/esm/entry.webpack';
+import ReactPlayer from 'react-player/file';
+import interviewerPDF from './pdf/1.pdf';
+import interviewerVideo from './pdf/video.mp4';
```
Comment (carmeltsadok) [inline src/pages/PreparationKit/index.jsx:5]: not exist, app not running because it can't resolve it

File: src/pages/PreparationKit/index.jsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    setPageNumber(1);
+    setPageAmount(numPages);
+  };
+
+  useEffect(() => {
+    setPageHeight(sidebarRef.current.getBoundingClientRect().height);
+  }, [sidebarRef.current?.getBoundingClientRect().height]);
+
+  return (
+    <div className={classes.root}>
```
Comment (carmeltsadok) [inline src/pages/PreparationKit/index.jsx:28]: wrap this component with `DashboardCard` to be matched with all pages's styling

File: src/pages/PreparationKit/index.jsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            Previous
+          </button>
+          <button
+            type='button'
+            disabled={pageNumber >= pageAmount}
+            onClick={() => changePage(1)}
+          >
+            Next
+          </button>
+        </div>
```
Comment (carmeltsadok) [inline src/pages/PreparationKit/index.jsx:66]: make it a bit prettier using material-ui components

File: src/pages/PreparationKit/index.jsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      </div>
+      <div className={classes.playerWrapper}>
+        <ReactPlayer
+          url={interviewerVideo}
+          className={classes.player}
+          controls
+        />
+      </div>
```
Comment (carmeltsadok) [inline src/pages/PreparationKit/index.jsx:74]: couldn't see it because of the import :)

File: src/pages/PreparationKit/index.jsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            of
+            {' '}
+            {pageAmount || '--'}
+          </p>
+          <button
+            type='button'
+            disabled={pageNumber <= 1}
+            onClick={() => changePage(-1)}
+          >
+            Previous
```
Comment (carmeltsadok) [inline src/pages/PreparationKit/index.jsx:57]: Change to hebrew

File: src/common/SideBarNav/SideBarNav.jsx [hunk 7]
```diff
@@ -32,9 +32,10 @@ import luzIconActive from '../../utils/images/aside/aside-luz-button-active.svg'
 import configApp from '../../appConf';
 import UserStoreInstance from '../../stores/User.store';
 
-const Sidebar = () => {
+const Sidebar = React.forwardRef((props, ref) => {
```
Comment (carmeltsadok) [inline src/common/SideBarNav/SideBarNav.jsx:35]: Why did you add ref?

File: src/pages/PreparationKit/index.jsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    setPageAmount(numPages);
+  };
+
+  useEffect(() => {
+    setPageHeight(sidebarRef.current.getBoundingClientRect().height);
+  }, [sidebarRef.current?.getBoundingClientRect().height]);
+
+  return (
+    <div className={classes.root}>
+      <div className={classes.PDFContainer}>
```
Comment (carmeltsadok) [inline src/pages/PreparationKit/index.jsx:29]: also, make the page the same height as the sidebar so there will be no scrolling

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: src/pages/PreparationKit/index.jsx (hunk 3 above)
Comment (PCOffline) [reply src/pages/PreparationKit/index.jsx]: Done

File: src/pages/PreparationKit/index.jsx (hunk 4 above)
Comment (PCOffline) [reply src/pages/PreparationKit/index.jsx]: Done

File: src/pages/PreparationKit/index.jsx (hunk 6 above)
Comment (PCOffline) [reply src/pages/PreparationKit/index.jsx]: Done

File: src/common/SideBarNav/SideBarNav.jsx (hunk 7 above)
Comment (PCOffline) [reply src/common/SideBarNav/SideBarNav.jsx:35]: Fixed

File: package.json (hunk 1 above)
Comment (PCOffline) [reply package.json]: Done

File: mockserver/preparation-kit/preparation-kit.manager.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import { dirname } from 'path';
```
Comment (carmeltsadok) [inline mockserver/preparation-kit/preparation-kit.manager.js:1]: rename file to `preparationKit.manager.js`

File: mockserver/preparation-kit/preparation-kit.router.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import express from 'express';
```
Comment (carmeltsadok) [inline mockserver/preparation-kit/preparation-kit.router.js:1]: rename file to `preparationKit.router.js`

File: src/pages/PreparationKit/index.jsx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+              disabled={pageNumber >= pageAmount}
+              onClick={() => changePage(1)}
+            >
+              <ArrowBackIcon />
+            </IconButton>
+          </div>
+        )}
+        <Document
+          file={`${configApp.apiUri}/api/preparation-kit/pdf`}
+          noData={<p>No File Chosen</p>}
```
Comment (carmeltsadok) [inline src/pages/PreparationKit/index.jsx:59]: maybe change ti hebrew

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: mockserver/preparation-kit/preparation-kit.manager.js (hunk 9 above)
Comment (carmeltsadok) [reply mockserver/preparation-kit/preparation-kit.manager.js:1]: Change folder's namer to `preparationKit`

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: mockserver/preparation-kit/preparation-kit.manager.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+export default class PreparationKitManager {
+  static getPDF(req, res) {
+    res.sendFile(`${__dirname}/files/samplePDF.pdf`);
+  }
+
+  static getVideo(req, res) {
+    res.sendFile(`${__dirname}/files/sampleVideo.mp4`);
+  }
```
Comment (carmeltsadok) [inline mockserver/preparation-kit/preparation-kit.manager.js:13]: add files to git

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: mockserver/preparation-kit/preparation-kit.manager.js (hunk 12 above)
Comment (carmeltsadok) [reply mockserver/preparation-kit/preparation-kit.manager.js:13]: about the pdf- make a mock like the video you sent me so it won't be free on the internet :)

File: src/pages/PreparationKit/index.styles.js [hunk 13]
```diff
@@ -0,0 +1,52 @@
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: '100%',
+    display: 'flex',
```
Comment (carmeltsadok) [inline src/pages/PreparationKit/index.styles.js:6]: add `alignItems: 'center',`

File: src/pages/PreparationKit/index.styles.js [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+    justifyContent: 'space-around',
+    overflow: 'auto',
+
+    [theme.breakpoints.down('md')]: {
+      flexDirection: 'column',
+      alignItems: 'stretch',
```
Comment (carmeltsadok) [inline src/pages/PreparationKit/index.styles.js:12]:
```suggestion
```

File: src/pages/PreparationKit/index.styles.js [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+      justifyContent: 'flex-start',
+    },
+  },
+
+  document: {
+    '& > canvas': { borderRadius: '15px' },
+  },
+
+  documentContainer: {
+    marginLeft: '10%',
```
Comment (carmeltsadok) [inline src/pages/PreparationKit/index.styles.js:22]:
```suggestion
    marginLeft: '10%',
    [theme.breakpoints.down('sm')]: {
      marginLeft: '0',
    },
```

File: src/pages/PreparationKit/index.styles.js [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+    position: 'relative',
+    padding: '2%',
+    background: theme.palette.secondary.light,
+    borderRadius: '15px',
+    marginTop: '1%',
+    alignSelf: 'center',
+
+    [theme.breakpoints.down('md')]: {
+      alignSelf: 'flex-start',
+    },
```
Comment (carmeltsadok) [inline src/pages/PreparationKit/index.styles.js:42]:
```suggestion
```

File: src/pages/PreparationKit/index.styles.js (hunk 14 above)
Comment (carmeltsadok) [reply src/pages/PreparationKit/index.styles.js:12]: Suggesting another styling in this file: centering the documents on small screen

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: mockserver/preparation-kit/preparation-kit.manager.js (hunk 9 above)
Comment (PCOffline) [reply mockserver/preparation-kit/preparation-kit.manager.js]: done

File: mockserver/preparation-kit/preparation-kit.router.js (hunk 10 above)
Comment (PCOffline) [reply mockserver/preparation-kit/preparation-kit.router.js]: done

File: src/pages/PreparationKit/index.jsx (hunk 11 above)
Comment (PCOffline) [reply src/pages/PreparationKit/index.jsx]: done

File: mockserver/preparation-kit/preparation-kit.manager.js (hunk 12 above)
Comment (PCOffline) [reply mockserver/preparation-kit/preparation-kit.manager.js]: done

File: src/pages/PreparationKit/index.styles.js (hunk 13 above)
Comment (PCOffline) [reply src/pages/PreparationKit/index.styles.js:6]: This breaks the arrows placement

File: src/pages/PreparationKit/index.styles.js (hunk 14 above)
Comment (PCOffline) [reply src/pages/PreparationKit/index.styles.js:12]: Adding `alignItems: center` breaks the placement and makes the document go outside of the screen

**Review by carmeltsadok — APPROVED** [review]

**Comment by lustigy96** [comment]
Why did you add such a big file to the mock server? it that neccesery? It explodes the client repo @carmeltsadok @maykremer @PCOffline

**Pull request merged**